### ISSUE TITLE
Server Node setup issue

### ISSUE THREAD

**Issue by slhck** [fg-inet/DASH-streaming-setup#3]
When running `vagrant up` I get:

```
==> server: Box 'bento/ubuntu-16.04' could not be found. Attempting to find and install...
    server: Box Provider: virtualbox
    server: Box Version: >= 0
==> server: Loading metadata for box 'bento/ubuntu-16.04'
    server: URL: https://vagrantcloud.com/bento/ubuntu-16.04
==> server: Adding box 'bento/ubuntu-16.04' (v202002.04.0) for provider: virtualbox
==> server: Importing base box 'bento/ubuntu-16.04'...
==> server: Matching MAC address for NAT networking...
==> server: Checking if box 'bento/ubuntu-16.04' version '202002.04.0' is up to date...
==> server: Setting the name of the VM: vagrant_files_server_1585485479465_92692
==> server: Fixed port collision for 22 => 2222. Now on port 2201.
==> server: Clearing any previously set network interfaces...
==> server: Preparing network interfaces based on configuration...
    server: Adapter 1: nat
    server: Adapter 2: hostonly
==> server: Forwarding ports...
    server: 22 (guest) => 2201 (host) (adapter 1)
==> server: Running 'pre-boot' VM customizations...
==> server: Booting VM...
==> server: Waiting for machine to boot. This may take a few minutes...
    server: SSH address: 127.0.0.1:2201
    server: SSH username: vagrant
    server: SSH auth method: private key
    server:
    server: Vagrant insecure key detected. Vagrant will automatically replace
    server: this with a newly generated keypair for better security.
    server:
    server: Inserting generated public key within guest...
    server: Removing insecure key from the guest if it's present...
    server: Key inserted! Disconnecting and reconnecting using new SSH key...
==> server: Machine booted and ready!
==> server: Checking for guest additions in VM...
    server: The guest additions on this VM do not match the installed version of
    server: VirtualBox! In most cases this is fine, but in rare cases it can
    server: prevent things such as shared folders from working properly. If you see
    server: shared folder errors, please make sure the guest additions within the
    server: virtual machine match the version of VirtualBox you have installed on
    server: your host and reload your VM.
    server:
    server: Guest Additions Version: 6.1.2
    server: VirtualBox Version: 6.0
==> server: Configuring and enabling network interfaces...
==> server: Mounting shared folders...
    server: /vagrant => /Users/werner/Documents/Projects/inet/DASH-streaming-setup/vagrant_files
    server: /home/vagrant/DASH-setup => /Users/werner/Documents/Projects/inet/DASH-streaming-setup/DASH-setup/server
==> server: Running provisioner: shell...
    server: Running: /var/folders/xr/f7sr_4_x33q4d51k3bh995080000gn/T/vagrant-shell20200329-38870-dpnrbg.sh
    server: WARNING:
    server: apt
    server:
    server: does not have a stable CLI interface.
    server: Use with caution in scripts.
    server: Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
    server: Get:2 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
    server: Get:3 http://security.ubuntu.com/ubuntu xenial-security InRelease [109 kB]
    server: Get:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
    server: Get:5 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [1,120 kB]
    server: Get:6 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [840 kB]
    server: Get:7 http://archive.ubuntu.com/ubuntu xenial-updates/main i386 Packages [907 kB]
    server: Get:8 http://archive.ubuntu.com/ubuntu xenial-updates/main Translation-en [427 kB]
    server: Get:9 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [793 kB]
    server: Get:10 http://archive.ubuntu.com/ubuntu xenial-updates/universe i386 Packages [719 kB]
    server: Get:11 http://security.ubuntu.com/ubuntu xenial-security/main i386 Packages [647 kB]
    server: Get:12 http://archive.ubuntu.com/ubuntu xenial-updates/universe Translation-en [331 kB]
    server: Get:13 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 Packages [16.8 kB]
    server: Get:14 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse i386 Packages [15.9 kB]
    server: Get:15 http://security.ubuntu.com/ubuntu xenial-security/main Translation-en [319 kB]
    server: Get:16 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [487 kB]
    server: Get:17 http://security.ubuntu.com/ubuntu xenial-security/universe i386 Packages [420 kB]
    server: Get:18 http://security.ubuntu.com/ubuntu xenial-security/universe Translation-en [199 kB]
    server: Get:19 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 Packages [5,728 B]
    server: Get:20 http://security.ubuntu.com/ubuntu xenial-security/multiverse i386 Packages [5,880 B]
    server: Fetched 7,580 kB in 3s (2,458 kB/s)
    server: Reading package lists...
    server: Building dependency tree...
    server: Reading state information...
    server: 27 packages can be upgraded. Run 'apt list --upgradable' to see them.
    server: ## Installing the NodeSource Node.js 10.x repo...
    server: ## Populating apt-get cache...
    server:
    server: + apt-get update
    server: Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
    server: Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
    server: Hit:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
    server: Hit:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
    server: Reading package lists...
    server:
    server: ## Confirming "xenial" is supported...
    server:
    server: + curl -sLf -o /dev/null 'https://deb.nodesource.com/node_10.x/dists/xenial/Release'
    server:
    server: ## Adding the NodeSource signing key to your keyring...
    server:
    server: + curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
    server: OK
    server:
    server: ## Creating apt sources list file for the NodeSource Node.js 10.x repo...
    server:
    server: + echo 'deb https://deb.nodesource.com/node_10.x xenial main' > /etc/apt/sources.list.d/nodesource.list
    server: + echo 'deb-src https://deb.nodesource.com/node_10.x xenial main' >> /etc/apt/sources.list.d/nodesource.list
    server:
    server: ## Running `apt-get update` for you...
    server:
    server: + apt-get update
    server: Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
    server: Hit:2 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
    server: Get:3 https://deb.nodesource.com/node_10.x xenial InRelease [4,584 B]
    server: Hit:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
    server: Hit:5 http://security.ubuntu.com/ubuntu xenial-security InRelease
    server: Get:6 https://deb.nodesource.com/node_10.x xenial/main amd64 Packages [765 B]
    server: Fetched 5,349 B in 0s (9,043 B/s)
    server: Reading package lists...
    server:
    server: ## Run `sudo apt-get install -y nodejs` to install Node.js 10.x and npm
    server: ## You may also need development tools to build native addons:
    server:      sudo apt-get install gcc g++ make
    server: ## To install the Yarn package manager, run:
    server:      curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
    server:      echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
    server:      sudo apt-get update && sudo apt-get install yarn
    server: WARNING:
    server: apt
    server:
    server: does not have a stable CLI interface.
    server: Use with caution in scripts.
    server: Reading package lists...
    server: Building dependency tree...
    server: Reading state information...
    server: The following NEW packages will be installed:
    server:   nodejs
    server: 0 upgraded, 1 newly installed, 0 to remove and 27 not upgraded.
    server: Need to get 16.2 MB of archives.
    server: After this operation, 81.1 MB of additional disk space will be used.
    server: Get:1 https://deb.nodesource.com/node_10.x xenial/main amd64 nodejs amd64 10.19.0-1nodesource1 [16.2 MB]
    server: dpkg-preconfigure: unable to re-open stdin: No such file or directory
    server: Fetched 16.2 MB in 5s (3,204 kB/s)
    server: Selecting previously unselected package nodejs.
    server: (Reading database ...
    server: (Reading database ... 5%
    server: (Reading database ... 10%
    server: (Reading database ... 15%
    server: (Reading database ... 20%
    server: (Reading database ... 25%
    server: (Reading database ... 30%
    server: (Reading database ... 35%
    server: (Reading database ... 40%
    server: (Reading database ... 45%
    server: (Reading database ... 50%
    server: (Reading database ... 55%
    server: (Reading database ... 60%
    server: (Reading database ... 65%
    server: (Reading database ... 70%
    server: (Reading database ... 75%
    server: (Reading database ... 80%
    server: (Reading database ... 85%
    server: (Reading database ... 90%
    server: (Reading database ... 95%
    server: (Reading database ... 100%
    server: (Reading database ...
    server: 38555 files and directories currently installed.)
    server: Preparing to unpack .../nodejs_10.19.0-1nodesource1_amd64.deb ...
    server: Unpacking nodejs (10.19.0-1nodesource1) ...
    server: Processing triggers for man-db (2.7.5-1) ...
    server: Setting up nodejs (10.19.0-1nodesource1) ...
    server: npm
    server:
    server: ERR!
    server:
    server: code
    server:  ENOENT
    server: npm
    server:
    server: ERR!
    server:
    server: syscall
    server:  open
    server: npm
    server:
    server: ERR!
    server:
    server: path
    server:  /home/vagrant/DASH-setup/node_modules/argparse/node_modules/sprintf-js/package.json.2910646466
    server: npm
    server:
    server: ERR!
    server:
    server: errno
    server:  -2
    server: npm
    server:
    server: ERR!
    server:
    server: enoent
    server:  ENOENT: no such file or directory, open '/home/vagrant/DASH-setup/node_modules/argparse/node_modules/sprintf-js/package.json.2910646466'
    server: npm
    server:
    server: ERR!
    server:
    server: enoent
    server:  This is related to npm not being able to find a file.
    server: npm
    server:
    server: ERR!
    server:
    server: enoent
    server:
    server: npm
    server:
    server: ERR!
    server:  A complete log of this run can be found in:
    server: npm
    server:
    server: ERR!
    server:      /root/.npm/_logs/2020-03-29T12_39_03_267Z-debug.log
    server:
    server: > metricexample@0.0.0 start /home/vagrant/DASH-setup
    server: > nodemon ./bin/www
    server: sh: 1:
    server: nodemon: not found
    server: npm
    server:
    server: ERR!
    server:
    server: code
    server:  ELIFECYCLE
    server: npm
    server:
    server: ERR!
    server:
    server: syscall
    server:  spawn
    server: npm
    server:
    server: ERR!
    server:
    server: file
    server:  sh
    server: npm ERR! errno ENOENT
    server: npm ERR! metricexample@0.0.0 start: `nodemon ./bin/www`
    server: npm ERR! spawn ENOENT
    server: npm ERR!
    server: npm ERR! Failed at the metricexample@0.0.0 start script.
    server: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
    server: npm
    server:
    server: ERR!
    server:  A complete log of this run can be found in:
    server: npm
    server:
    server: ERR!
    server:      /root/.npm/_logs/2020-03-29T12_39_03_923Z-debug.log
```

Vagrant 2.2.5 under macOS…

**Comment by slhck** [comment]
Similar error on the client:

```
    client: Setting up nodejs (10.19.0-1nodesource1) ...
    client: npm ERR! code
    client:  ENOENT
    client: npm ERR! syscall open
    client: npm ERR! path /home/vagrant/DASH-setup/node_modules/argparse/node_modules/sprintf-js/package.json.3474474763
    client: npm ERR! errno -2
    client: npm ERR! enoent ENOENT: no such file or directory, open '/home/vagrant/DASH-setup/node_modules/argparse/node_modules/sprintf-js/package.json.3474474763'
    client: npm
    client:  ERR!
    client:  enoent
    client:  This is related to npm not being able to find a file.
    client: npm ERR!
    client: enoent
    client:
    client: npm
    client:  ERR!
    client:  A complete log of this run can be found in:
    client: npm ERR!     /root/.npm/_logs/2020-03-29T12_37_41_009Z-debug.log
```

**Comment by slhck** [comment]
I will try with Linux later or tomorrow, but I suspect it is because of some outdated package lock file.

**Comment by PolynomialDivision** [comment]
Thanks for reporting. We fixed everything, see c82b29e450cad23b87766a44dea1f8f196b81f32 for more details.
